app/controllers/hosts_controller.rb

```diff
@@ -448,7 +448,7 @@ def submit_rebuild_config
   end
 
   def submit_multiple_build
-    @hosts.delete_if do |host|
+    @hosts.to_a.delete_if do |host|
```

Comment:
Yup they fail indeed - this happens because @hosts in here is a FriendlyId relation (which shares properties with arrays and you can use delete_if on them) but on Rails 4 it's a plain ActiveRecord relation. This all stem from the fact resource_base is a ActiveRecord relation.

I'll write a couple of tests for these actions and update the pull request with that.